### ISSUE TITLE
examples are not built by default

### ISSUE THREAD

**Issue by litaochen** [AdaptiveParticles/LibAPR#106]
In the instruction there is a statement:
```
This will create the libapr.so library in the build directory, as well as all of the examples.
```
But When I follow the instruction, the examples were not built.
By inspecting the file "CMakeList.txt" I saw the switch is "OFF" by default.
```
option(APR_BUILD_EXAMPLES "Build APR examples" OFF)
```
Shall it be changed to "ON"?

**Comment by cheesema** [comment]
Thanks i'll follow up on this and make sure the README is correct.

**Comment by krzysg** [comment]
I think that CMakeLists.txt is OK right now - this is library so default settings should reflect that and build only lib. Any additional stuff like examples, tests, wrappers etc. should be explicitly requested by user by providing -DAPR_BUILD_EXAMPLES=ON etc. options to cmake.

**Comment by cheesema** [comment]
Yip we just need to update the README as it currenlty says:

>
mkdir build
cd build
cmake ..
make
This will create the libapr.so library in the build directory, as well as all of the examples.
>

**Comment by vrutten** [comment]
Hi,
I'm trying to build the code (on OSX).
I ran the instructions:
`mkdir build
cd build
cmake ..
make`

The targets are built but no 'libapr.so' is created.

The files which are generate are:
- Makefile
- libapr.1.dylib
- libapr.dylib
- external
- libapr.1.2.0.dylib
- ConfigAPR.h
- CMakeFiles
- CMakeCache.txt
- cmake_install.cmake

Any ideas as to what could be going wrong? 


Looking forward to testing the libraries!

**Comment by cheesema** [comment]
I vrutten, apologies for the delay! Didn't see this question on the existing issue.

Thanks for picking this up, their was simply a typo on the OSX build instructions. On OSX, the built library target is the libapr.dylib (the .so is for linux). This has been fixed in #121
